### ISSUE TITLE
[TF2] Since last ALSA update the game crashes

### ISSUE THREAD

**Issue by tupo2** [ValveSoftware/Source-1-Games#5188]
Worked fine before Steam update 23/09/11.

```
Steam Version:  1694466342
Steam Client Build Date:  Mon, Sep 11 10:49 PM UTC +01:00
Steam Web Build Date:  Sun, Sep 10 1:01 AM UTC +01:00
Steam API Version:  SteamClient020
```

Errors in the terminal:
```
ERROR: ld.so: object '/home/user/.steam/debian-installation/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
/home/user/.steam/debian-installation/steamapps/common/Team Fortress 2/hl2.sh: line 72: 37971 Segmentation fault      ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"
(process:37978): GLib-GObject-CRITICAL **: 10:52:40.514: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```

That's what I tried:
* Cleaned caches.
* Disabled Steam Overlay.
* Tested with Steam Client Beta.
* Tested with other games (they are launched and playable).
* Reinstalled Steam and TF2 in new folders (using default settings).
* Tested with another Linux native and Source game: Codename CURE (it crashes also, same errors as TF2).

**Comment by nanopone** [comment]
nearly the same thing here on fedora 38 with the steam flatpak (although i had the exact same problem with the rpmfusion package from fedora)

when using steam linux runtime:
`/bin/bash: /usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/libtinfo.so.6: no version information available (required by /bin/bash)
pid 1304 != 1300, skipping destruction (fork without exec?)
pid 1309 != 1300, skipping destruction (fork without exec?)
/bin/bash: /usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/libtinfo.so.6: no version information available (required by /bin/bash)
/bin/bash: /usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/libtinfo.so.6: no version information available (required by /bin/bash)
pid 1344 != 1300, skipping destruction (fork without exec?)
src/tcmalloc.cc:278] Attempt to free invalid pointer 0x9145ad0 
/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Team Fortress 2/hl2.sh: line 72:  1346 Aborted                 (core dumped) ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"`

when not using steam linux runtime:
`ERROR: ld.so: object '/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pid 1407 != 1406, skipping destruction (fork without exec?)
ERROR: ld.so: object '/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
src/tcmalloc.cc:278] Attempt to free invalid pointer 0xa81dad0 
/home/rae/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Team Fortress 2/hl2.sh: line 72:  1409 Aborted                 (core dumped) ${GAME_DEBUGGER} "${GAMEROOT}"/${GAMEEXE} "$@"`

**Comment by kisak-valve** [comment]
Hello @nanopone, you've described the issue being tracked at #5043.

**Comment by tupo2** [comment]
It seems to be due to an update of libasound2 (I'm using ALSA) from version 1.2.9 to 1.2.10.
All other apps and games works fine, except Source games.
Using another sound server can be used as workaround, but a fix would be appreciated.

**Comment by avlec** [comment]
I tried downgrading to alsa-lib and lib32-alsa-lib to 1.2.9 and it didn't seem to change anything.

**Comment by tupo2** [comment]
This issue is fixed with libasound2 1.2.10-2 in Debian and should be with alsa-lib 1.2.10-2 in Arch Linux.
